src/sagas/export-to-codesandbox.saga.js

```diff
@@ -0,0 +1,236 @@
+  const result = yield call([dialog, dialog.showMessageBox], {
+    type: 'warning',
+    title: 'Exported code will be public',
+    message:
+      'By deploying to CodeSandbox, the code of your project will be made public.',
+    buttons: ['OK, export project', 'Cancel'],
+  });
+
+  if (result === 1) {
+    yield put(exportToCodesandboxFinish(projectId));
+    // abort export
+    yield cancel();
+  }
+
+  // deploying to existing sandbox not supported by CLI
+  // --> deploy option is picked by default and there is no option to overwrite an existing sandbox.
+  // if (codesandboxUrl) {
+  //   // deploy to existing sandbox --> todo add a way to create a new sandbox
+  //   args.push(['deploy', codesandboxUrl]);
+  // }
+
+  const channel = yield call(
+    spawnProcessChannel,
+    formatCommandForPlatform('codesandbox'),
+    args,
+    projectPath,
+    'CODESANDBOX:EXPORT'
+  );
+
+  yield call(watchExportMessages, channel, projectId);
+}
+
+export function* watchExportMessages(channel: any, projectId: string): any {
+  let output;
+  var currentStep = 0;
+  try {
+    while (true) {
+      output = yield take(channel);
+      if (!output.hasOwnProperty('exit')) {
+        const next = yield call(checkMessage, output, currentStep);
+
+        // Modify/increment currentStep if requested by checkMessage
+        // will be skipped if message requires no action.
+        if (next) {
+          if (next === 'inc') {
+            currentStep++;
+          } else if (!isNaN(next)) {
+            // numeric value
+            currentStep = next;
+          }
+        }
+      } else {
+        // Yield exit code and complete stdout
+        if (output.data && output.data.stdout.includes('[error]')) {
+          // e.g. errors:
+          // - This project is too big, it contains 140 files which is more than the max of 120. (exported Gatsby project)
+          // - This project is too big, it contains 56 directories which is more than the max of 50.
+          yield call([dialog, dialog.showMessageBox], {
+            type: 'error',
+            title: 'Codesandbox Export Error',
+            // $FlowFixMe
+            message: /\[error\] (.*)/.exec(output.data.stdout)[1],
+          });
+          yield put(exportToCodesandboxFinish(projectId));
+        }
+        yield output;
+
+        // Close channel manually --> emitter(END) inside spwanProcessChannel would exit too early
+        channel.close();
+      }
+    }
+  } finally {
+  }
+}
+
+// exportSteps are the messages & responses for exporting to Codesandbox (same order as they occur in terminal)
+export const exportSteps = [
+  { msg: 'Do you want to sign in using GitHub', response: 'Y' },
+  {
+    // Would be good to not go to codesandbox as the user already entered the token
+    // we need to open codesandbox because otherwise there won't be a question for the token
+    // --> current CLI api doesn't support what we need here. A command line arg for the token would help.

```

Comment:
I need to double check this with a fresh state. I think this behaviour changed during my work on it.
I think it's working as expected and not opening a page to codesandbox but I have to check this.